### PR TITLE
feat: add basic slash command

### LINKED ISSUE #1: Setup environment
- [x]  Version managements
    - [x]  Add virtualenv and requirements.txt file
- [x]  README file
    - [x]  Make basic. We will not publish code open-source before our public docs are published.
- [x]  Alchemy ORM for DB
- [x] gitignore
- [x] Understand bot abilities
  - [x] receiving parameters
  - [x] slashes
  - [x] working with button, images

### PR DESCRIPTION
**Pull request by marrakonda**
- slash commands are added
- welcome dm added
- button, input, and image examples added
- pre-commit hook added to format files with black automatically on commit
- changeling generation added

closes #1

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -4,4 +4,6 @@ SQLAlchemy==2.0.23
 discord.py==2.3.2
```
Comment (sche) [inline requirements.txt:4]: can we remove discord.py?

File: .env.example [hunk 2]
```diff
@@ -1 +1,2 @@
 DISCORD_TOKEN=<TOKEN>
+DISCORD_GUILD=1179544636499431445
```
Comment (sche) [inline .env.example:2]: Is it Goverland id?

Comment (sche) [reply .env.example:2]: I think it is better to make it as `<GUILD>` placeholder

File: src/run.py [hunk 3]
```diff
@@ -1,31 +1,115 @@
...
-        print(f"We have logged in as {bot.user}")
-        for guild in bot.guilds:
-            print(f"Guild {guild}. ID: {guild.id}")
-        print("All channels:")
-        for channel in bot.get_all_channels():
-            print(channel)
-
-    bot.run(os.environ["DISCORD_TOKEN"])
+        print(
+            f"{bot.me.name} is connected to the following guild:\n"
```
Comment (sche) [inline src/run.py:21]: Will it potentially show all the guilds/servers that connected this bot?

File: src/run.py [hunk 4]
```diff
@@ -1,31 +1,115 @@
...
+            f"{bot.guilds[0].name}(id: {bot.guilds[0].id})"
+        )
+
+    @bot.event
+    async def on_member_join(member):
```
Comment (sche) [inline src/run.py:26]: Will it be displayed for all new users joining Goverland server?
The scope of this bot should be limited to Server admins only. Use case: an admin adds the bot, configures the channel where updates will be coming to, adds/removes DAOs to follow.

File: src/run.py [hunk 5]
```diff
@@ -1,31 +1,115 @@
...
+        await member.create_dm()
+        await member.dm_channel.send(
+            f"Hi {member.name}, welcome to {bot.guilds[0].name} 🤖! All you have to do is type a slash “/” to bring "
+            f"up a list of commands that Goverland Bot can do!"
+        )
+
+    @bot.event
+    async def on_error(event, *args):
```
Comment (sche) [inline src/run.py:34]: Could you please add a short comment / function doc describing a use case when it might happen?

File: src/run.py [hunk 6]
```diff
@@ -1,31 +1,115 @@
...
+                required=True,
+            ),
+        ],
+    )
+    async def search_dao(ctx: interactions.CommandContext, text: str):
+        await ctx.send(f"You are searching for '{text}'!")
+
+    @bot.command(
+        name="button_test",
+        description="This is the first command I made!",
```
Comment (sche) [inline src/run.py:67]: :)

File: src/run.py [hunk 7]
```diff
@@ -1,31 +1,115 @@
...
+    )
+    async def button_test(ctx):
+        await ctx.send("testing", components=ButtonHello)
+
+    @bot.component("hello")
```
Comment (sche) [inline src/run.py:72]: How about placing `"hello"` into a variable. It will reduce potential errors.

**Review by sche — CHANGES_REQUESTED** [review]
Good job! 👏
Let's make a call and discuss it with a live demo.

File: requirements.txt (hunk 1 above)
Comment (marrakonda) [reply requirements.txt:4]: Done

File: src/run.py (hunk 3 above)
Comment (marrakonda) [reply src/run.py:21]: Changed it to show all of them

File: src/run.py (hunk 4 above)
Comment (marrakonda) [reply src/run.py:26]: This is a nice way to say hi to every joiner of a guild. Nothing more will be available for a non-admin user.

File: src/run.py (hunk 5 above)
Comment (marrakonda) [reply src/run.py:34]: Done

File: src/run.py (hunk 7 above)
Comment (marrakonda) [reply src/run.py:72]: I did it a little bit different and more robust :)

File: src/run.py [hunk 8]
```diff
@@ -1,31 +1,135 @@
...
-from discord.ext import commands
-from src.commands import test, hybrid
+import interactions
+from interactions import ButtonStyle
+
+from src.application.components import ComponentsService, ButtonAction
 
 load_dotenv()
+DISCORD_TOKEN = os.getenv("DISCORD_TOKEN")
+DISCORD_GUILD = int(os.getenv("DISCORD_GUILD"))
```
Comment (sche) [inline src/run.py:11]: Did we decide to remove DISCORD_GUILD last time?

File: src/run.py (hunk 4 above)
Comment (sche) [reply src/run.py:28]: So non-admin users will not see anything among bot slash commands?

Comment (sche) [reply src/run.py:28]: I see later in code. Question closed.

File: src/run.py [hunk 9]
```diff
@@ -1,31 +1,135 @@
...
+        Privacy Policy: https://www.goverland.xyz/privacy
+        """
+
+        # button_link ("About Goverland")
+        # button_link ("Privacy Policy")
+        await ctx.send(message)
+
+    # --- ADMIN ---
+
+    @bot.command(
```
Comment (sche) [inline src/run.py:82]: I didn't test it, but I don't understand from the code how Admin commands are different from Regular user commands.

**Review by sche — APPROVED** [review]

**Pull request merged**